backend/app/api/endpoints/openapi_responses.py

```
@@ -484,21 +1121,117 @@ async def cancel_response(
+            await task_kinds_service.cancel_task(
+                db=db,
+                task_id=task_id,
+                user_id=current_user.id,
+                background_task_runner=background_tasks.add_task,
+            )
+        except HTTPException as e:
+            if e.status_code == 404:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail=f"Response '{response_id}' not found",
+                )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)